src/components/datepicker/js/datepickerDirective.js

```diff
@@ -677,28 +677,29 @@
     if (this.isCalendarOpen) {
       var self = this;
 
+      self.detachCalendarPane();
+      self.ngModelCtrl.$setTouched();
+
+      self.documentElement.off('click touchstart', self.bodyClickHandler);
```

Comment:
Mainly because they don't _need_ to be in it. The only one that should fire on the next tick is the `calendarOpen` since it prevents the calendar reopening immediately if the user turned on `openOnFocus`.